object_model/artifact.py

```diff
@@ -365,57 +365,179 @@ def pull(self, manifest={}):
 
         self.__commit__()
 
-    def peek(self, head=25, taip='default', func = lambda x: x):
-        trueVersioningDir = self.xp_state.versioningDirectory
-        self.xp_state.versioningDirectory = os.path.expanduser('~') + '/' + '1fdf8583bfd663e98918dea393e273cc'
 
-        # Create temporary versioning directory to store files
-        if os.path.exists(self.xp_state.versioningDirectory):
-            rmtree(self.xp_state.versioningDirectory)
-            os.mkdir(self.xp_state.versioningDirectory)
+    def peek(self, head=25, manifest={}, bindings = {}, func = lambda x: x):
+
+        self.xp_state.versioningDirectory = os.path.expanduser('~') + '/' + 'flor.d'
+
+        # Recreate the tmpexperiment directory
+        tmpexperiment = self.xp_state.tmpexperiment
+        if os.path.exists(tmpexperiment):
+            rmtree(tmpexperiment)
+            os.mkdir(tmpexperiment)
         else:
-            os.mkdir(self.xp_state.versioningDirectory)
+            os.mkdir(tmpexperiment)
 
-        # Update elements visited and literals array
         self.xp_state.visited = []
         util.activate(self)
+        userDefFiles = set(os.listdir()) - self.xp_state.ghostFiles
+        experimentName = self.xp_state.EXPERIMENT_NAME
 
-        if taip == "default":
-            try:
-                # Just pull once and put output in folder peek0 within versioning directory.
-                original_dir = os.getcwd()
-                dst = self.xp_state.versioningDirectory + '/' + "peek0"
-                copytree(os.getcwd(), dst, True)
-                os.chdir(dst)
-                self.__pull__()
+        try:
+            old_literal_bindings = {}
+
+            # Iterate through bindings, store old bindings and reset to new bindings for pull
+            for literal, binding in bindings.items():
+                if literal.__oneByOne__:
+                    old_literal_bindings[literal] = literal.v[0]
+                    literal.v[0] = binding
+                else:
+                    old_literal_bindings[literal] = literal.v
+                    literal.v = binding
+
+
+            # If there are literals that are unbound, then set them to default.
+            if len(bindings) != len(self.__getLiteralsAttached__()):
+                unbound_literals = [literal for literal in self.xp_state.literals if literal not in bindings.keys()]
```

Comment:
is `self.xp_state.literals` the same as `literalsAttached`? How did you confirm this?